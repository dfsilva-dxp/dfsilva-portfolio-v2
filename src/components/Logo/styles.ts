import styled, { css } from "styled-components";

export const Wrap = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray50};
    position: relative;
    z-index: ${theme.layers.base};

    .blend-mode {
      mix-blend-mode: exclusion;
    }

    .text {
      display: block;
    }

    svg {
      width: 7.8125rem;
    }

    @media (max-width: 768px) {
      .text {
        display: none;
      }

      svg {
        width: 2.5rem;
      }
    }
  `}
`;
