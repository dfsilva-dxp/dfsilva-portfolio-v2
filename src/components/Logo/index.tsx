import * as S from "./styles";

const Logo = () => {
  return (
    <>
      <S.Desktop>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="108"
          height="30"
          fill="none"
          aria-label="Daniel Silva Desktop logo"
        >
          <path
            className="text"
            fill="currentColor"
            d="M12.758 27.293v2.391h3.36V8.118h-3.36v8.557c-1.21-1.744-2.966-2.677-5.115-2.677C3.33 13.998 0 17.848 0 22.014c0 4.286 3.224 7.971 7.643 7.971 2.15 0 3.89-.947 5.115-2.692Zm-9.413-5.294c0-2.677 2.088-4.857 4.813-4.857 2.709 0 4.797 2.195 4.797 4.857 0 2.677-2.088 4.828-4.797 4.828-2.725 0-4.813-2.136-4.813-4.828ZM21.457 17.472h2.906v12.212h3.36V17.472h3.527v-3.188h-3.512v-2.256c0-.677.44-1.113 1.12-1.113 1.03 0 1.862.617 2.392 1.444V8.704c-.59-.616-1.53-.962-2.377-.962-2.482 0-4.51 1.744-4.51 4.271v2.271h-2.906v3.188ZM45.457 18.164l1.166-3.008c-1.499-.677-3.179-1.188-4.828-1.188-2.876 0-6.039 1.324-6.039 4.617 0 5.084 7.87 4.151 7.87 6.708 0 1.218-1.801 1.549-2.724 1.549-1.272 0-2.74-.602-3.86-1.158l-1.286 2.962c1.574.767 3.39 1.354 5.16 1.354 2.846 0 6.055-1.308 6.055-4.587 0-5.249-7.84-4.136-7.84-6.738 0-1.308 1.695-1.534 2.679-1.534 1.24 0 2.558.467 3.647 1.023ZM53.263 29.684h3.375v-15.4h-3.375v15.4Zm0-17.746h3.375v-3.82h-3.375v3.82ZM63.612 29.684h3.36V8.118h-3.36v21.566Z"
          />
          <path
            fill="#1BFAC1"
            d="m84.04 14.284-4.026 10.858-4.026-10.858h-3.602l6.07 15.4h3.102l6.069-15.4H84.04Z"
          />
          <path
            className="text"
            fill="currentColor"
            d="M104.059 27.293v2.391h3.36v-15.4h-3.36v2.391c-1.21-1.744-2.966-2.677-5.115-2.677-4.314 0-7.643 3.85-7.643 8.016 0 4.286 3.224 7.971 7.643 7.971 2.149 0 3.89-.947 5.115-2.692Zm-9.413-5.294c0-2.677 2.088-4.857 4.813-4.857 2.709 0 4.797 2.195 4.797 4.857 0 2.677-2.088 4.828-4.797 4.828-2.725 0-4.813-2.136-4.813-4.828Z"
          />
          <path
            fill="#1BFAC1"
            d="m73.033.25-.448 2.942c-.022.146.048.289.167.339L87.27 9.659c.15.064.314-.047.342-.232l.449-2.942c.022-.145-.049-.289-.168-.339L73.376.018c-.15-.063-.314.047-.342.232Z"
          />
        </svg>
      </S.Desktop>
      <S.Mobile>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="16"
          height="30"
          fill="none"
          aria-label="Daniel Silva Mobile Logo"
        >
          <path
            fill="#1BFAC1"
            d="M11.654 14.284 7.628 25.142 3.602 14.284H0l6.069 15.4h3.103l6.069-15.4h-3.587Z"
          />
          <path
            fill="#1BFAC1"
            d="M11.654 14.284 7.628 25.142 3.602 14.284H0l6.069 15.4h3.103l6.069-15.4h-3.587ZM.647.25.2 3.192c-.023.146.048.289.167.339l14.518 6.128c.15.064.313-.047.342-.232l.448-2.942c.023-.145-.048-.289-.167-.339L.99.018C.839-.045.675.065.647.25Z"
          />
        </svg>
      </S.Mobile>
    </>
  );
};

export default Logo;
