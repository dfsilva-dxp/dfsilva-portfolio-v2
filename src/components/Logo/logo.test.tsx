import { screen } from "@testing-library/dom";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Logo from ".";

describe("<Logo/>", () => {
  it("should render currectly logo", () => {
    const { container } = renderWithTheme(<Logo />);

    expect(
      screen.getByLabelText(/Daniel Silva Desktop logo/i).parentElement
    ).toHaveStyle({
      color: "#E1E1E6",
      position: "relative",
      "z-index": 10
    });

    expect(container).toMatchSnapshot();
  });
});
