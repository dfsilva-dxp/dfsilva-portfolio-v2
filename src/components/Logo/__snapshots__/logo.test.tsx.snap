// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> should render currectly logo 1`] = `
.c0 {
  color: #E1E1E6;
  position: relative;
  z-index: 10;
  display: none;
  -webkit-transition: display 0.25s linear;
  transition: display 0.25s linear;
}

.c0 svg {
  width: 7.8125rem;
}

.c1 {
  color: #E1E1E6;
  position: relative;
  z-index: 10;
  display: block;
  -webkit-transition: display 0.25s linear;
  transition: display 0.25s linear;
}

@media (min-width:768px) {
  .c0 {
    display: block;
  }
}

@media (min-width:768px) {
  .c1 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <svg
      aria-label="Daniel Silva Desktop logo"
      fill="none"
      height="30"
      width="108"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.758 27.293v2.391h3.36V8.118h-3.36v8.557c-1.21-1.744-2.966-2.677-5.115-2.677C3.33 13.998 0 17.848 0 22.014c0 4.286 3.224 7.971 7.643 7.971 2.15 0 3.89-.947 5.115-2.692Zm-9.413-5.294c0-2.677 2.088-4.857 4.813-4.857 2.709 0 4.797 2.195 4.797 4.857 0 2.677-2.088 4.828-4.797 4.828-2.725 0-4.813-2.136-4.813-4.828ZM21.457 17.472h2.906v12.212h3.36V17.472h3.527v-3.188h-3.512v-2.256c0-.677.44-1.113 1.12-1.113 1.03 0 1.862.617 2.392 1.444V8.704c-.59-.616-1.53-.962-2.377-.962-2.482 0-4.51 1.744-4.51 4.271v2.271h-2.906v3.188ZM45.457 18.164l1.166-3.008c-1.499-.677-3.179-1.188-4.828-1.188-2.876 0-6.039 1.324-6.039 4.617 0 5.084 7.87 4.151 7.87 6.708 0 1.218-1.801 1.549-2.724 1.549-1.272 0-2.74-.602-3.86-1.158l-1.286 2.962c1.574.767 3.39 1.354 5.16 1.354 2.846 0 6.055-1.308 6.055-4.587 0-5.249-7.84-4.136-7.84-6.738 0-1.308 1.695-1.534 2.679-1.534 1.24 0 2.558.467 3.647 1.023ZM53.263 29.684h3.375v-15.4h-3.375v15.4Zm0-17.746h3.375v-3.82h-3.375v3.82ZM63.612 29.684h3.36V8.118h-3.36v21.566Z"
        fill="#fff"
      />
      <path
        d="m84.04 14.284-4.026 10.858-4.026-10.858h-3.602l6.07 15.4h3.102l6.069-15.4H84.04Z"
        fill="#1BFAC1"
      />
      <path
        d="M104.059 27.293v2.391h3.36v-15.4h-3.36v2.391c-1.21-1.744-2.966-2.677-5.115-2.677-4.314 0-7.643 3.85-7.643 8.016 0 4.286 3.224 7.971 7.643 7.971 2.149 0 3.89-.947 5.115-2.692Zm-9.413-5.294c0-2.677 2.088-4.857 4.813-4.857 2.709 0 4.797 2.195 4.797 4.857 0 2.677-2.088 4.828-4.797 4.828-2.725 0-4.813-2.136-4.813-4.828Z"
        fill="#fff"
      />
      <path
        d="m84.04 14.284-4.026 10.858-4.026-10.858h-3.602l6.07 15.4h3.102l6.069-15.4H84.04ZM73.033.25l-.448 2.942c-.022.146.048.289.167.339L87.27 9.659c.15.064.314-.047.342-.232l.449-2.942c.022-.145-.049-.289-.168-.339L73.376.018c-.15-.063-.314.047-.342.232Z"
        fill="#1BFAC1"
      />
    </svg>
  </div>
  <div
    class="c1"
  >
    <svg
      aria-label="Daniel Silva Mobile Logo"
      fill="none"
      height="30"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.654 14.284 7.628 25.142 3.602 14.284H0l6.069 15.4h3.103l6.069-15.4h-3.587Z"
        fill="#1BFAC1"
      />
      <path
        d="M11.654 14.284 7.628 25.142 3.602 14.284H0l6.069 15.4h3.103l6.069-15.4h-3.587ZM.647.25.2 3.192c-.023.146.048.289.167.339l14.518 6.128c.15.064.313-.047.342-.232l.448-2.942c.023-.145-.048-.289-.167-.339L.99.018C.839-.045.675.065.647.25Z"
        fill="#1BFAC1"
      />
    </svg>
  </div>
</div>
`;
